firmware/boron-ins-fsm/src/im21door.cpp

```diff
@@ -70,9 +72,18 @@ doorData checkIM21(){
     // read as: doorLowBatteryFlag is true if doorStatus AND 0b0100 is not 0b0000
     doorLowBatteryFlag = (currentDoorData.doorStatus & (1 << 2)) != 0;
 
+    // After a certain thereshold, the next door message received will trigger a boron heartbeat
+    if (millis() - doorLastMessage >= MSG_TRIGGER_SM_HEARTBEAT_THRESHOLD){
+      doorHeartbeatReceivedFlag = true;
+    }
+
+    //record the time a IM21 message was received
+    doorLastMessage = millis();
+
     // Checks if the 3rd bit of doorStatus is 1
     if ((currentDoorData.doorStatus & (1 << 3)) != 0){
       doorHeartbeatReceived = millis();
+      doorHeartbeatReceivedFlag = true;
```

Comment:
Do we still need this one? Or can we get away with just having the earlier check of whether it's been longer than the `MSG_TRIGGER_SM_HEARTBEAT_THRESHOLD`?

In which case, this one we could change the name of to be more all-encompassing, like `doorMessageReceivedFlag`.

Reply:
Hmm it depends on what we would like to do. 
Removing this would result in a boron heartbeat 
1. once every 10 mins
2. if we receive any IM21 door broadcast after 15 mins of not receiving anything

whereas with it, it result in a boron heartbeat
1. once every 10 mins
2. if we receive any IM21 door broadcast after 15 mins of not receiving anything
3. Whenever we receive a IM21 heartbeat
I can see both of them being alright, but to save broadcast, Im gonna implement removing it.